consensus/istanbul/core/final_committed.go

```diff
@@ -21,6 +21,6 @@ import "github.com/ethereum/go-ethereum/common"
 func (c *core) handleFinalCommitted() error {
 	logger := c.logger.New("state", c.state)
 	logger.Trace("Received a final committed proposal")
-	c.startNewRound(common.Big0)
+	go c.startNewRound(common.Big0)
```

Comment:
Why do we need a goroutine that was not there before?

Reply:
During network transition, the legacy IBFT network is first stopped before starting QBFT consensus. As part of stopping legacy IBFT network, the engine is stopped as well and all the events are unsubscribed. Stopping of the engine is blocked until `core/handler.go::handleEvents()` is exited. The goroutine above makes sure `handleEvents()` exits due to unsubscribing of events and legacy IBFT engine is stopped